internal/controller/ansibleRun/ansibleRun.go

```diff
@@ -185,7 +188,13 @@ func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.E
 		}
 	}
 	if buff.Len() != 0 {
-		if err := c.fs.WriteFile(filepath.Join(dir, runnerutil.Hosts), buff.Bytes(), 0600); err != nil {
+		if err := c.fs.WriteFile(filepath.Join(dir, runnerutil.Hosts), buff.Bytes(), inventoryPerm); err != nil {
+			return nil, fmt.Errorf("%s %s: %w", errWriteInventory, runnerutil.Hosts, err)
+		}
+		// WriteFile only sets permissions for new files, do an explicit chmod to ensure changing permissions are updated
+		// on existing files
+		err := os.Chmod(filepath.Join(dir, runnerutil.Hosts), inventoryPerm)
+		if err != nil {
 			return nil, fmt.Errorf("%s %s: %w", errWriteInventory, runnerutil.Hosts, err)
```

Comment:
`errChmodInventory`

Reply:
Thanks, added an explicit error for this case